src/js/server/devpmt-datasources.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Devpmt Data Sources
+ *
+ * Data sources for retrieving directory GPII solutions registry entries,
+ * generic settings metadata, and GPII demo persona descriptions.
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+
+/**
+ * PreFetch DataSource
+ * `gpii.devpmt.dataSource.prefetchDataSource`
+ *
+ * A `kettle.dataSource` like component that is backed by a
+ * specific dataSource subcomponent. When the prefetchDataSource is
+ * created, it immediately fetches a copy of data using the dataSource's
+ * get method and stores it in the `current` member.  Each time the
+ * get method is used, the value of `current` is updated.
+ */
+fluid.defaults("gpii.devpmt.dataSource.prefetchDataSource", {
```

Comment:
Why do we have this prefetching?

Reply:
It's so that the datasource is fetched immediately once upon instantiation, rather than waiting for any type of manual call to it's `get` invoker.

Reply:
But why? How is it being used here?

Reply:
Well, currently it's just being used to the load the common terms files... there were more resources it was loading previously.  I suppose it could probably be replaced with a Kettle ResourceLoader?

Reply:
Yes, but why does it (attempt to) prefetch?

Reply:
So the resource is already available upon instantiation without having to invoke a get on it after it's been created.

I'm pretty sure this is actually exactly what the ResourceLoader is for, I just wasn't aware of it when I first created this last year.

Reply:
The problem is, it won't be, since you don't do anything to wait for the data to arrive.

Reply:
I guess yeah, if a request was immedatiately made it might not be loaded yet.  Either way, I should elimiate all of this in favor of using the ResourceLoader correct?

Reply:
Yes - also the ResourceLoader will be the primary means of consuming I/O in the upcoming framework - at least certainly for "once per component" I/O

Reply:
All of these custom datasources have been removed.